File: py/1822.sign-of-the-product-of-an-array.py
#
# @lc app=leetcode id=1822 lang=python3
#
# [1822] Sign of the Product of an Array
#

# @lc code=start
class Solution:
    def arraySign(self, nums: List[int]) -> int:
        res = 1
        for n in nums:
            if n < 0:
                res *= -1
            elif n == 0:
                return 0
        return res
# @lc code=end

File: cpp/1822.sign-of-the-product-of-an-array.cpp
/*
 * @lc app=leetcode id=1822 lang=cpp
 *
 * [1822] Sign of the Product of an Array
 */
#include <vector>
using namespace std;
// @lc code=start
class Solution
{
public:
    int arraySign(vector<int> &nums)
    {
        int res = 1;
        for (auto n : nums)
        {
            if (n < 0)
            {
                res *= -1;
            }
            else if (n == 0)
            {
                return 0;
            }
        }
        return res;
    }
};
// @lc code=end
